### PR TITLE
Ci workflow rounda

### PR DESCRIPTION
**Pull request by jasonmcintosh**
An initial release flow.  
* Changes default make to use a docker builder and generates a full release.  NOT sure if this is the best path or not and need to cleanup the makefile, but... kids are bugging me... so later.  NOTE... I find using this pattern MUCH cleaner for those who have different golang or other configs to deal with AND it's idempotent build wise ;) 
* Publishes files to a GH release on main (still runs tests here... )
* Validates a PR by doing a make/build/test in a docker image

### CODE REVIEW

**Comment by jasonmcintosh** [comment]
@cla-bot check

File: .github/workflows/ci-build-test.yaml
Comment (megamorf) [inline .github/workflows/ci-build-test.yaml:34]: This way of passing information between workflow steps is deprecated. Use the newer format instead. See https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

File: Dockerfile.build
Comment (megamorf) [inline Dockerfile.build:4]: Running mkdir is unnecessary. `WORKDIR` creates the path if it doesn't exist.

**Review by megamorf — CHANGES_REQUESTED** [review]
I don't know how I feel about the this implementation yet. I think there's a lot of room for improvement but they heavily depend on the way CI and the release process should look like.

**Assumptions**

As a user of ae I expect:
- multi-arch binaries + checksums to be available for releases
- there to be a multi-arch Docker image for ae
  - the image to be available on GHCR and Docker Hub

***

My personal preference based on my experience is to be platform agnostic and to not rely to heavily on CI system features. So building on top of make and Docker seems like the way to go.

I think it makes sense to put more emphasis on Docker layer caching and build artifact caching, see:
- https://medium.com/@tonistiigi/faster-multi-platform-builds-dockerfile-cross-compilation-guide-part-1-ec087c719eaf
- https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#run---mounttypecache

I think there is another blog post from Tonis Tiigi where he covers the cache mount type in more detail.

**Comment by jasonmcintosh** [comment]
Keep in mind:
* This is potentially temporary or can be replaced by the goreleaser based build instead.
* I like the docker builder - it can already support multiple release formats so... this is just the "compiler" piece for CI integration, or for local integration. That said, be nice to make it cleaner in a future update.  Is that needed in this PR or an additional PR?  I tend to prefer small fast PR's & incremental improvements.
* I could see other pieces being cache... open to improvements - or other alternatives, this starts the ball rolling on #6

File: .github/workflows/ci-build-test.yaml
Comment (krisnova) [inline .github/workflows/ci-build-test.yaml:1]: Can we use the word `Main` instead of `Master` please?

**Comment by krisnova** [comment]
Hey 👋🏻 checking in. Can you help me understand how this relates to #22

**Comment by jasonmcintosh** [comment]
SO kinda waiting on that other PR before I'd update or merge this PR.  KEY pieces on THIS that may be useful is changing the Make & local builds to use a Docker based builds instead of the workspace installed golang.  This gives you consistent builds no matter what environment, plugins, tools, etc. and true cross platform builds in an isolated environment per build. 

NOTE if we don't want that kinda pattern, happy to just close this in favor of the other - I've just seen this pattern used pretty regularly to handle inconsistencies between different end users for multi user builds.  Example being I can't right now checkout and run make as i don't run golang 19 locally due to some other work been doing without adjusting my installed toolset.

**Comment by jasonmcintosh** [comment]
Going to close this PR for now in favor of goreleaser, but if there's interest in making local builds more... "idempotent"? :P or easier for those without the base golang stuff installed, the makefile changes are here in the fork for reference

**Pull request closed without merging**
